### PR TITLE
Added new FileUploadV2 function to avoid server side file timeouts

### PR DESCRIPTION
**Pull request by sarthakkothari**
### Linking issue 
This PR solves for - https://github.com/slack-go/slack/issues/1108
Implements the methods suggested in - https://github.com/slackapi/python-slack-sdk/releases/tag/v3.19.0

Need to account for nil interface because slack doesn't give `SlackResponse` or any json response for that matter when uploading file via post to [GetUploadURLExternal](https://api.slack.com/methods/files.getUploadURLExternal)'s UploadURL

### CODE REVIEW

**Comment by sarthakkothari** [comment]
@kanata2 making a PR for the first time, how do I request your review/add label?

**Comment by kanata2** [comment]
@sarthakkothari Sorry for delay. Will review within a few days. 🙏

File: files.go [hunk 1]
```diff
@@ -416,3 +455,119 @@ func (api *Client) ShareFilePublicURLContext(ctx context.Context, fileID string)
...
+		return nil, err
+	}
+	if response.Err() != nil {
+		return nil, response.Err()
+	}
+	return response, nil
+}
+
+// UploadFileV2 uploads file to a given slack channel using 3 steps -
+// 	1. Get an upload URL using files.getUploadURLExternal API
```
Comment (kanata2) [inline files.go:527]:
```suggestion
//  1. Get an upload URL using files.getUploadURLExternal API
```

File: files.go [hunk 2]
```diff
@@ -145,6 +146,44 @@ type ListFilesParameters struct {
 	Cursor  string
 }
 
+type FileUploadV2Parameters struct {
```
Comment (kanata2) [inline files.go:149]: [IMO] Prefer `UploadFileV2Parameters`. It would be easier to use if aligned with the method name.

File: files.go [hunk 3]
```diff
@@ -145,6 +146,44 @@ type ListFilesParameters struct {
...
+	SnippetText     string
+}
+
+type getUploadURLExternalResponse struct {
+	UploadURL string `json:"upload_url"`
+	FileID    string `json:"file_id"`
+	SlackResponse
+}
+
+type uploadToExternalParams struct {
```
Comment (kanata2) [inline files.go:169]: `uploadToURLParameters` ?

File: files.go [hunk 4]
```diff
@@ -416,3 +455,119 @@ func (api *Client) ShareFilePublicURLContext(ctx context.Context, fileID string)
 	}
 	return &response.File, response.Comments, &response.Paging, nil
 }
+
+// getUploadURLExternal gets a URL and fileID from slack which can later be used to upload a file
+func (api *Client) getUploadURLExternal(ctx context.Context, fileSize int, fileName, altText, snippetText string) (*getUploadURLExternalResponse, error) {
```
Comment (kanata2) [inline files.go:460]: How about defining struct since there are many parameters?

File: files.go [hunk 5]
```diff
@@ -416,3 +455,119 @@ func (api *Client) ShareFilePublicURLContext(ctx context.Context, fileID string)
...
+	} else if params.File != "" {
+		err = postLocalWithMultipartResponse(ctx, api.httpclient, params.UploadURL, params.File, "file", api.token, values, nil, api)
+	} else if params.Reader != nil {
+		err = postWithMultipartResponse(ctx, api.httpclient, params.UploadURL, params.Filename, "file", api.token, values, params.Reader, nil, api)
+	}
+	return err
+}
+
+// completeUploadExternal once files are uploaded, this completes the upload and shares it to the specified channel
+func (api *Client) completeUploadExternal(ctx context.Context, fileID string, params FileUploadV2Parameters) (file *completeUploadExternalResponse, err error) {
```
Comment (kanata2) [inline files.go:497]: It is preferable to define struct instead of using `FileUploadV2Prameters`.

**Pull request merged**